### DISCUSSION TITLE
TableRowCollection.Update - unnecessary copying of data?

### DISCUSSION THREAD

**Discussion by m5x** [spectreconsole/spectre.console#814] [Q&A]
Is there a reason for replacing items in List<> instances in method [TableRowCollection.Update](https://github.com/spectreconsole/spectre.console/blob/52c1d9122bce4c73ae9205e2f2aff01cba03a4c4/src/Spectre.Console/Widgets/Table/TableRowCollection.cs#L123) by removing old item and then inserting new one at the same index? This makes List<> do unnecessary copying which could be avoided by simply replacing the old item with new one (i.e. list[index] = newItem).

**Comment by patriksvensson** [comment]
Not a bug, moving this to discussions.

**Reply by patriksvensson** [reply]
I have no idea. There is probably a reason, but maybe not.
Send a PR in which you provide a benchmark, and we will know for sure.

**Reply by m5x** [reply]
Is benchmark really needed? It's kind of obvious.

**Reply by phil-scott-78** [reply]
@m5x, I think what Patrik is saying is that if this is a perf issue you are seeing, we are happy to take a PR, but with it not being anything we are encountering that PR would be nice to have tests and some benchmarks to go along with it.

**Reply by m5x** [reply]
No sweat. We evidently have different opinions about what is acceptable in good quality code. I would call such usage of List<> "programmer's faux pas" and fix it immediately while you don't even seem to see it. I guess that makes further discussion on this topic pointless.
